### ISSUE TITLE
[BUG] BlobServiceClient constructor fails with AzureCoreEventSource exception on Android

### ISSUE THREAD

**Issue by tschramme86** [Azure/azure-sdk-for-net#24127]
**Describe the bug**
Given this line of code:
```
var service = new BlobServiceClient($"BlobEndpoint={myBlobEndpoint};SharedAccessSignature={myBlobSAS}");
```

When running locally in Windows Environment (e.g. in Console or Unit Test Application) it runs fine and I can use the `service` object afterwards to access my Blobs successfully. But when I run exactly the same line on Android (Xamarin.Forms) an `AzureCoreEventSource` exception is thrown from the constructor and my application crashes.

When I now explicitly disable logging, it works fine also in Android Application:
```
var options = new BlobClientOptions { Diagnostics = { IsLoggingEnabled = false } };
var service = new BlobServiceClient($"BlobEndpoint={myBlobEndpoint};SharedAccessSignature={myBlobSAS}", options);
```

**Expected behavior**
Creating a BlobServiceClient is also possible in a Xamarin.Forms application on Android without explicitly disable logging. Default configurations should not lead to strange exceptions.

**Actual behavior (include Exception or Stack Trace)**
When running on Android without explicitly disabled logging, the constructor throws the following exception and the application crashes.
```
System.TypeInitializationException: The type initializer for 'Azure.Core.Diagnostics.AzureCoreEventSource' threw an exception. 
---> System.NotImplementedException: The method or operation is not implemented.  at System.Diagnostics.Tracing.EventSource.GetSources () [0x00000] in /Users/builder/jenkins/workspace/archive-mono/2020-02/android/release/mcs/class/corlib/System.Diagnostics.Tracing/EventSource.cs:318   at Azure.Core.Diagnostics.AzureEventSource.DeduplicateName (System.String eventSourceName) [0x00010] in <0aa90fe0646d47bc95bff520717dbf16>:0   at Azure.Core.Diagnostics.AzureEventSource..ctor (System.String eventSourceName) [0x00000] in <0aa90fe0646d47bc95bff520717dbf16>:0   at Azure.Core.Diagnostics.AzureCoreEventSource..ctor () [0x00000] in <0aa90fe0646d47bc95bff520717dbf16>:0   at Azure.Core.Diagnostics.AzureCoreEventSource..cctor () [0x00000] in <0aa90fe0646d47bc95bff520717dbf16>:0    --- End of inner exception stack trace ---  at Azure.Core.Pipeline.LoggingPolicy..cctor () [0x00000] in <0aa90fe0646d47bc95bff520717dbf16>:0

```

**To Reproduce**
Try to create a BlobServiceClient instance in an Xamarin.Forms / Xamarin application on Android.

When no explicit logging behavior is configured, the constructor crashes. When logging is explicitly disabled via `BlobClientOptions` everything works fine

**Environment:**
 - Azure.Storage.Blobs 12.10.0 with Azure.Core 1.19.0
 - Xamarin.Forms 5.0.0.2125 Application on Android 11 Emulator / Android 9 Device
 - Visual Studio 2019 16.11.3

**Comment by jsquire** [comment]
Hi @tschramme86.  Thank you for your feedback and we regret that you're experiencing difficulties.   This issue was first reported in #24055, and was fixed by #24097.   It will be released in early October, during our next release cycle.

In the meantime, downgrading to `Azure.Storage.Blobs` v12.9.1 is the recommended work-around, as the bug was introduced in the next version.  If you're unable to downgrade and like to test the fix, it is available in our [nightly feed](pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-net/nuget/v3/index.json).

**Comment by jsquire** [comment]
//fyi: @pakrym
